### PR TITLE
update mysql known compatibility link in readme.md

### LINKED ISSUE #10059: Known Compatibility page not working
This page 
https://github.com/pingcap/tidb/blob/master/sql/mysql-compatibility.md
on 
https://github.com/pingcap/tidb

is not found

also, auto increment counter supported? would like to chk out the difference between the two

### PR DESCRIPTION
**Pull request by AmazingTurtle**
Fixes #10059

### What problem does this PR solve? 
https://github.com/pingcap/tidb/issues/10059

### What is changed and how it works?
Link update in readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ TiDB ("Ti" stands for Titanium) is an open-source NewSQL database that supports
 
 - __MySQL Compatible Syntax__
 
-    TiDB acts like it is a MySQL 5.7 server to your applications. You can continue to use all of the existing MySQL client libraries, and in many cases, you will not need to change a single line of code in your application. Because TiDB is built from scratch, not a MySQL fork, please check out the list of [known compatibility differences](./sql/mysql-compatibility.md).
+    TiDB acts like it is a MySQL 5.7 server to your applications. You can continue to use all of the existing MySQL client libraries, and in many cases, you will not need to change a single line of code in your application. Because TiDB is built from scratch, not a MySQL fork, please check out the list of [known compatibility differences](https://github.com/pingcap/docs/blob/master/sql/mysql-compatibility.md).
```
Comment (zz-jason) [inline README.md]: I clicked this link on the README page of TiDB master https://github.com/pingcap/tidb, it redirects me to https://www.pingcap.com/docs/sql/mysql-compatibility/
@lilin90 Is this behavior expected?

Comment (lilin90) [reply README.md]: Yes. I think it is expected. @zz-jason

**Comment by queenypingcap** [comment]
@lilin90 PTAL

**Comment by lilin90** [comment]
I suggest closing this PR because the link issue has been fixed in https://github.com/pingcap/tidb/pull/10140. @zz-jason @AmazingTurtle

**Comment by AmazingTurtle** [comment]
Already done in another PR

**Pull request closed without merging**